sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala

```diff
@@ -77,4 +77,29 @@ package object expressions  {
     /** Uses the given row to store the output of the projection. */
     def target(row: MutableRow): MutableProjection
   }
+
+  /**
+   * A utility classes for check the expression equality in semantic, not literally.
+   */
+  trait ExpressionEquals
```

Comment:
What is the point of this trait?  Creating a class hierarchy where you don't need one just adds complexity to the code.